src/workerd/io/worker-interface.capnp

```diff
@@ -168,6 +168,20 @@ interface HibernatableWebSocketEventDispatcher {
   # Run a hibernatable websocket event
 }
 
+struct JsValue {
+  v8Serialized @0 :Data;
+  # JS value that has been serialized for network transport.
+}
+
+interface JsRpcTarget {
+  call @0 (methodName :Text, serializedArgs :JsValue) -> (result :JsValue);
```

Comment:
It looks like as the code is currently written, the `result` of `call()` will either be the return value or an exception, and the `outcome` result of `getJsRpcTarget` signals whether an exception was thrown. Thus in order to correctly interpret the results of `call()` you must also know the separate `outcome`.

This seems awkward and seems in particular like it means `JsRpcTarget` cannot be reused later to represent nested objects. We should instead have the result of `call()` directly indicate if it's an exception. I'd recommend declaring it like:

```
call @0 (methodName :Text, serializedArgs :JsValue) -> CallResults;

struct CallResults {
  union {
    value @0 :JsValue;
    # Returned a value.

    exception @1 :JsValue;
    # Exception thrown from JS code. (Note that call() may also throw a KJ exception, which should be converted to JS as usual.)
  }
}
```

Reply:
Thanks for the suggestion, this feels a lot simpler IMO. I'll look into getting rid of the separate `outcome` in the morning, there's a decent amount of plumbing for it so I want to think about if we still  want it for anything before I yank it out https://github.com/cloudflare/workerd/pull/1028/commits/74726536df31113a419860ef405d195d2c081164